kafka-connect-s3/src/test/java/io/confluent/connect/s3/integration/S3SinkConnectorIT.java

```diff
@@ -0,0 +1,288 @@
+  }
+
+  /**
+   * Creates root client that will be used to change bucket permissions.
+   * Prerequisite : Access key and Secret access key should be set as environment variables
+   */
+  private void createS3RootClient() {
+    s3RootClient = AmazonS3ClientBuilder.standard()
+      .withCredentials(
+          new AWSStaticCredentialsProvider(
+              new BasicAWSCredentials(
+                System.getenv("ROOT_USER_ACCESS_KEY_ID"),
+                System.getenv("ROOT_USER_SECRET_ACCESS_KEY"))))
+      .withRegion("ap-south-1")
+      .build();
+  }
+
+  @After
+  public void close() {
+    //delete the test bucket
+    deleteBucket(S3_TEST_BUCKET_NAME);
+    stopConnect();
+  }
+
+  /**
+   * Success scenario : End to end test
+   * @throws Throwable
+   */
+  @Test
+  public void testToAssertConnectorAndDestinationRecords() throws Throwable {
+
+    // create topics in Kafka
+    KAFKA_TOPICS.forEach(topic -> connect.kafka().createTopic(topic, 1));
+    // send records to kafka
+    sendRecordsToKafka();
+
+    Map<String, String> props = getConnectorProps();
+    props.put("s3.bucket.name", S3_TEST_BUCKET_NAME);
+
+    // start a sink connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+    // wait for tasks to spin up
+    int minimumNumTasks = Math.min(KAFKA_TOPICS.size(), TASKS_MAX);
+
+    waitForConnectorToStart(CONNECTOR_NAME, minimumNumTasks);
+    waitForConnectorToCompleteSendingRecords(totalNoOfRecordsProduced, FLUSH_SIZE, S3_TEST_BUCKET_NAME);
+
+    // assert records
+    assertEquals(totalNoOfRecordsProduced/FLUSH_SIZE, getNoOfObjectsInS3(S3_TEST_BUCKET_NAME));
+  }
+
+  /**
+   * Case in which bucket permissions/policies are changed while uploading records
+   * Prerequisite : Access key and Secret access key should be set as environment variables
+   * Blocker : Assert Condition
+   * @throws Exception
+   */
+  @Test
+  public void testWithRevokedWritePermissions() throws Exception {
+
+    addReadWritePolicyToBucket(S3_TEST_BUCKET_NAME);
+    KAFKA_TOPICS.forEach(topic -> connect.kafka().createTopic(topic, 1));
+
+    // send records to kafka
+    sendRecordsToKafka();
+
+    Map<String, String> props = getConnectorProps();
+    props.put("s3.bucket.name", S3_TEST_BUCKET_NAME);
+    props.put("aws.access.key.id", System.getenv("SECONDARY_USER_ACCESS_KEY_ID"));
+    props.put("aws.secret.access.key", System.getenv("SECONDARY_USER_SECRET_ACCESS_KEY"));
+
+    // start a sink connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+    // wait for tasks to spin up
+    int minimumNumTasks = Math.min(KAFKA_TOPICS.size(), TASKS_MAX);
+
+    waitForConnectorToStart(CONNECTOR_NAME, minimumNumTasks);
+    Thread.sleep(5000);
+
+    // revoke read/write permission
+    s3RootClient.deleteBucketPolicy(S3_TEST_BUCKET_NAME);
+
+    // produce more records to kafka
+    sendRecordsToKafka();
+    Thread.sleep(10000);
+  }
+
+  @Test
+  public void testWithNetworkInterruption() throws Throwable {
```

Comment:
This is one way of testing it. Another test should be added with pumba which makes the squid service intermittently and we should still be able to get all the records in S3. Refer to Elastic Search IT for more details.